src/components/modal/delete-modal.tsx

```diff
@@ -0,0 +1,115 @@
+import * as React from 'react';
+import {
+  k8sDeleteResource,
+  K8sModelCommon,
+  K8sResourceCommon,
+} from '@openshift/dynamic-plugin-sdk-utils';
+import {
+  Button,
+  ButtonType,
+  ButtonVariant,
+  Form,
+  ModalVariant,
+  Stack,
+  StackItem,
+  Text,
+  TextContent,
+  TextVariants,
+  ValidatedOptions,
+} from '@patternfly/react-core';
+import { Formik } from 'formik';
+import { InputField } from '../../shared';
+import { ComponentProps, createModalLauncher } from './createModalLauncher';
+
+type DeleteResourceModalProps = ComponentProps & {
+  obj: K8sResourceCommon;
+  model: K8sModelCommon;
+  description?: string;
```

Comment:
The name of the resource will not be shown in bold if its a string.
```suggestion
  description?: React.ReactNode;
```